specs/NestedFrame.md

```diff
@@ -0,0 +1,268 @@
+    childFrame.NavigationCompleted += HandleChildFrameNavigationCompleted;
+    childFrame.Destroyed += HandleChildFrameDestroyed;
+}
+
+void HandleChildFrameDestroyed(object sender, object args) {
+    CoreWebView2Frame frame = (CoreWebView2Frame)sender;
+    MessageBox.Show(this, "Id: " + frame.FrameId + " FrameDestroyed", "Child frame Destroyed", MessageBoxButton.OK);
+}
+
+void HandleChildFrameNavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs args)
+{
+    CoreWebView2Frame frame = (CoreWebView2Frame)sender;
+    MessageBox.Show(this, "Id: " + frame.FrameId + " NavigationStarting", "Child frame Navigation", MessageBoxButton.OK);
+}
+
+void HandleChildFrameContentLoading(object sender, CoreWebView2ContentLoadingEventArgs args)
+{
+    CoreWebView2Frame frame = (CoreWebView2Frame)sender;
+    MessageBox.Show(this, "Id: " + frame.FrameId + " ContentLoading", "Child frame Content Loading", MessageBoxButton.OK);
+}
+
+void HandleChildFrameDOMContentLoaded(object sender, CoreWebView2DOMContentLoadedEventArgs args)
+{
+    CoreWebView2Frame frame = (CoreWebView2Frame)sender;
+    MessageBox.Show(this, "Id: " + frame.FrameId + " DOMContentLoaded", "Child frame DOM Content Loaded", MessageBoxButton.OK);
+}
+
+void HandleChildFrameNavigationCompleted(object sender, CoreWebView2NavigationCompletedEventArgs args)
+{
+    CoreWebView2Frame frame = (CoreWebView2Frame)sender;
+    MessageBox.Show(this, "Id: " + frame.FrameId + " NavigationCompleted", "Child frame Navigation Completed", MessageBoxButton.OK);
+}
+```
+
+# API Details
+## C++
+```
+/// Receives `FrameCreated` events.
+interface ICoreWebView2FrameNestedFrameCreatedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Frame* sender,
+      [in] ICoreWebView2FrameCreatedEventArgs* args);
+}
+
+/// This is the ICoreWebView2Frame interface.
+interface ICoreWebView2Frame7 : IUnknown {
+  /// Adds an event handler for the `FrameCreated` event.
+  /// Raised when a new direct descendant iframe is created.
+  /// Handle this event to get access to ICoreWebView2Frame objects.
+  /// Use `ICoreWebView2Frame.add_Destroyed` to listen for when this
+  /// iframe goes away.
+  /// 
+  /// \snippet ScenarioWebViewEventMonitor.cpp AddFrameCreated
+  HRESULT add_FrameCreated(
+      [in] ICoreWebView2FrameNestedFrameCreatedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_FrameCreated`.
+  HRESULT remove_FrameCreated(
+      [in] EventRegistrationToken token);
+}
+```
+
+C#
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Frame
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Frame7")]
+        {
+            event Windows.Foundation.TypedEventHandler<CoreWebView2Frame, CoreWebView2FrameCreatedEventArgs> FrameCreated;
+        }
+    }
+}
+```

```

Comment:
IIRC we had discussed that this addition would have impact on existing WebView2 APIs where various events can now sort of bubble from one corewebview2frame to their parent to their parent ... to the corewebview2. And how that interacts with event args for cancellation. This should probably be discussed somehow in this API spec. If it doesn't make sense to include in the API ref docs for CoreWebView2Frame.FrameCreated then perhaps you can add it to a new 'Appendix' section at the bottom that lists the other changes you plan to make to the other ref docs of impacted APIs.